src/thread/pthread/SDL_systhread.c

```diff
@@ -286,11 +286,12 @@ SDL_SYS_SetThreadPriority(SDL_ThreadPriority priority)
 void
 SDL_SYS_WaitThread(SDL_Thread * thread)
 {
-    pthread_join(thread->handle, 0);
+    if (thread->state.value != SDL_THREAD_STATE_ZOMBIE) {
```

Comment:
The thread state variable is set using SDL_AtomicCAS and should be read with SDL_AtomicGet
`if (SDL_AtomicGet(&thread->state.value) == SDL_THREAD_STATE_ZOMBIE) {`